GG/GG/RichText/RichText.h

```diff
@@ -0,0 +1,82 @@
+/** \file RichText.h \brief Contains the RichText class that allows you
+ to intersperse text with images.*/
+
+#ifndef RICHTEXT_H
+#define RICHTEXT_H
+
+#include <GG/ClrConstants.h>
+#include <GG/TextControl.h>
+
+#include <GG/RichText/BlockControl.h>
+
+namespace GG
+{
+
+/** \brief A control for showing text and images.
+ */
+class GG_API RichText: public Control
+{
+public:
+    typedef std::map<std::string, std::string> TAG_PARAMS;
+
+    //! An interface for object that create block controls from tags.
+    class IBlockControlFactory
+    {
+    public:
+        //! Creates a control from the tag (with unparsed parameters) and the content between the tags.
+        //! You own the returned control.
+        virtual BlockControl* CreateFromTag ( const std::string& tag,
+                                              const TAG_PARAMS& params,
+                                              const std::string& content,
+                                              const boost::shared_ptr<Font>& font,
+                                              Clr color,
+                                              Flags<TextFormat> format ) = 0;
+    };
+
+    //! The type of the object where we store control factories of tags.
+    typedef std::map<std::string, IBlockControlFactory*> BLOCK_FACTORY_MAP;
+
+    //! The special tag that is used to represent plaintext.
+    // Allows you to register a custom control for displaying plaintext.
+    static const std::string PLAINTEXT_TAG;
```

Comment:
The use can override the plaintext handler, as is done in the encyclopedia to enable links.